cdap-api/src/main/java/co/cask/cdap/internal/io/SQLSchemaParser.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.io;
+
+import co.cask.cdap.api.data.schema.Schema;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Parses a SQL like schema into a {@link Schema}.
+ */
+public final class SQLSchemaParser {
+  private String schema;
+  private int pos;
+  private int end;
+  // used so record names are unique
+  private int recordNum;
+
+  public SQLSchemaParser(String schema) {
+    // replace all whitespace with a single space
+    this.schema = schema.trim().replaceAll("\\s+", " ").toLowerCase();
```

Comment:
You don't need this if you have something like `String getNextToken()` (similar to the advancePastComma).